src/components/sidebar.js

```diff
@@ -112,15 +122,17 @@ let Sidebar = class Sidebar extends React.Component {
     //   )
     // }
 
+    console.log(this.props.selectedFeature);
+
     return (
       <PanelContainer>
-        <SidebarButton/>
+        <SidebarButton />
         <div className={`sidebar-container`}>
-            <SASearchField />
-          {this.props.selectedFeature ?
-            <TopPanel /> :
-            ""
-          }
+          <SASearchField />
+          {this.props.selectedFeature ? <TopPanel /> : <>
+            <Modal />
+            <Legend/>
```

Comment:
Legend should stay in Main, as moving Legend out of Main and into .sidebar-container means it disappears when there's a selection.

The first solution that comes to mind for keeping Modal from visually overlapping Legend while Legend is in Main is to place an empty <div /> in .sidebar-container with the same height as Legend and to make sure Legend's z-index is in front of .sidebar-container.

Reply:
I can make the legend stay rendered when a feature is selected if we want, I intentionally made it disappear since I thought that was the goal behavior. If it needs to be absolutely positioned outside then just using margin at the bottom that is as large as the legend should do it.

Reply:
Good idea. A bottom margin on .sidebar-container when there's no selected feature would be cleaner.

The desired behavior for the legend is in the first few screen of "v4 / Clickable Prototype" in Figma:
- [No selection, welcome visible](https://www.figma.com/file/vP3gxGjxW34i5gNMpu6sDW/Design-V1?node-id=1517%3A15227)
- [No selection, welcome dismissed](https://www.figma.com/file/vP3gxGjxW34i5gNMpu6sDW/Design-V1?node-id=1517%3A15234)
- [Selection, sidebar not collapsed](https://www.figma.com/file/vP3gxGjxW34i5gNMpu6sDW/Design-V1?node-id=1517%3A15279)
- [Selection, sidebar collapsed](https://www.figma.com/file/vP3gxGjxW34i5gNMpu6sDW/Design-V1?node-id=1517%3A15279)

Reply:
This is a potentially best of both worlds solution, it now renders as it did before when sidebar is opened/collapsed. At the same time it renders in such a way when no feature is selected that if we resize it down the line the rest of the sidebar will auto-adjust and we won't have to remember to change the bottom margin on the sidebar

Reply:
That works for me if you can get the welcome box to clip/overflow: scroll its contents when the window height is short instead of letting the Legend fall off the page.

Also, some stuff I noticed:
- There should be 5px more space between the search field and welcome box.
- Legend will need its own `pointer-events: auto` as being in the floating sidebar means it ignores clicks by default.
- The welcome box is on top of the search field's suggestions. The welcome box doesn't seem to actually need its z-index definition.

Reply:
Fixes applied, I'm used to working with ui frameworks like material ui or ant to handle flexbox aspects so it took a bit to manually control some aspects of the flexbox using actual div elements but it should function properly.

Reply:
Yeah, flexbox takes some getting used to, but it's pretty powerful. I had a few other comments, but it took as long to make the changes as it would have to write them up. Could you review the extra commits I've added?

Reply:
They look good, only question is about bringing back modal.js. Didn't seem like it was very generic or re-usable so to not confuse people down the line into thinking it's a general purpose modal component maybe we should move it to a deprecated folder or leave it in a branch we can reference later if we need it.

Reply:
If Maita's unpushed work on the hamburger menu implements modal dialogs the way I imagine she will, it'll be slightly more complicated for her to rebase her work onto master if modal.js disappears. If she doesn't end up using it, I'll ask her to delete it as part of the PR for her work.